notesapi/v1/views.py

```diff
@@ -275,20 +283,36 @@ def post(self, *args, **kwargs):  # pylint: disable=unused-argument
 
         Returns 400 request if bad payload is sent or it was empty object.
         """
-        if 'id' in self.request.data:
+        if not self.request.data or 'id' in self.request.data:
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
         try:
+            if Note.objects.filter(
+                    user_id=self.request.data['user'], course_id=self.request.data['course_id']
+            ).count() >= settings.MAX_ANNOTATIONS_PER_COURSE:
+                raise AnnotationsLimitReachedError
+
             note = Note.create(self.request.data)
             note.full_clean()
         except ValidationError as error:
             log.debug(error, exc_info=True)
             return Response(status=status.HTTP_400_BAD_REQUEST)
+        except AnnotationsLimitReachedError:
+            error_message = ugettext_noop(
```

Comment:
@ehteshamkafeel I think we don't need to use `ugettext_noop` because we not using `error_message` for any intermediate purpose and just returning in response.